narwhals/_polars/expr.py

```diff
@@ -20,22 +20,22 @@
 
 class PolarsExpr:
     def __init__(
-        self, expr: pl.Expr, dtypes: DTypes, backend_version: tuple[int, ...]
+        self: Self, expr: pl.Expr, dtypes: DTypes, backend_version: tuple[int, ...]
     ) -> None:
         self._native_expr = expr
         self._implementation = Implementation.POLARS
         self._dtypes = dtypes
         self._backend_version = backend_version
 
-    def __repr__(self) -> str:  # pragma: no cover
+    def __repr__(self: Self) -> str:  # pragma: no cover
         return "PolarsExpr"
 
-    def _from_native_expr(self, expr: pl.Expr) -> Self:
+    def _from_native_expr(self: Self, expr: pl.Expr) -> Self:
         return self.__class__(
             expr, dtypes=self._dtypes, backend_version=self._backend_version
         )
 
-    def __getattr__(self, attr: str) -> Any:
+    def __getattr__(self: Self, attr: str) -> Any:
```

Comment:
@MarcoGorelli this was the critical one! The other `__getattr__` are fine, sorry for that